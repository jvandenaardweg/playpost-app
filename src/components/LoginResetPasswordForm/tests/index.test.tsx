import React from 'react';
import { render, RenderAPI, fireEvent } from 'react-native-testing-library';

import { LoginResetPasswordForm } from '../index';

describe('LoginResetPasswordForm', () => {
  describe('rendering', () => {
    let wrapper: RenderAPI;
    const onChangeTextHandler = jest.fn();
    const onPressUpdatePasswordHandler = jest.fn();

    beforeEach(() => {
      wrapper = render(
        <LoginResetPasswordForm
          onChangeText={onChangeTextHandler}
          onPressUpdatePassword={onPressUpdatePasswordHandler}
          password=""
          resetPasswordToken=""
          isLoading={false}
          isSuccess={false}
        />
      );
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('interacting', () => {
    let wrapper: RenderAPI;
    const onChangeTextHandler = jest.fn();
    const onPressUpdatePasswordHandler = jest.fn();

    beforeEach(() => {
      wrapper = render(
        <LoginResetPasswordForm
          onChangeText={onChangeTextHandler}
          onPressUpdatePassword={onPressUpdatePasswordHandler}
          password=""
          resetPasswordToken=""
          isLoading={false}
          isSuccess={false}
        />
      );
    });

    it('should fire onPressUpdatePassword when the "reset my password" button is pressed', () => {
      fireEvent.press(wrapper.getByTestId('LoginResetPasswordForm-Button-update-password'));
      expect(onPressUpdatePasswordHandler).toHaveBeenCalledTimes(1);
    });

    it('should fire onChangeText when filling in the reset token TextInput', () => {
      const exampleResetToken = '123ABC';
      const resetTokenTextInput = wrapper.getByTestId('LoginResetPasswordForm-TextInput-reset-password-code');

      fireEvent.changeText(resetTokenTextInput, exampleResetToken);
      expect(onChangeTextHandler).toHaveBeenCalledWith('resetPasswordToken', exampleResetToken);
    });

    it('should fire onChangeText when filling in the password TextInput', () => {
      const examplePassword = 'p@ssWord!';
      const passwordTextInput = wrapper.getByTestId('LoginResetPasswordForm-TextInput-password');

      fireEvent.changeText(passwordTextInput, examplePassword);
      expect(onChangeTextHandler).toHaveBeenCalledWith('password', examplePassword);
    });

    it('should disable the buttons if isLoading is true', () => {
      wrapper.update(
        <LoginResetPasswordForm
          onChangeText={onChangeTextHandler}
          onPressUpdatePassword={onPressUpdatePasswordHandler}
          password=""
          resetPasswordToken=""
          isLoading={true}
          isSuccess={false}
        />
      );

      expect(wrapper.getByTestId('LoginResetPasswordForm-Button-update-password').props.disabled).toBe(true);
    });

    it('should disable the buttons if isSuccess is true', () => {
      wrapper.update(
        <LoginResetPasswordForm
          onChangeText={onChangeTextHandler}
          onPressUpdatePassword={onPressUpdatePasswordHandler}
          password=""
          resetPasswordToken=""
          isLoading={false}
          isSuccess={true}
        />
      );

      expect(wrapper.getByTestId('LoginResetPasswordForm-Button-update-password').props.disabled).toBe(true);
    });
  });
});
