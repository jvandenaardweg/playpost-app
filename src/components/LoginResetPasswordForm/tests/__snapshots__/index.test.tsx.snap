// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginResetPasswordForm rendering should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F1F1F1",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="LoginResetPasswordForm"
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#222222",
            "fontSize": 15,
            "lineHeight": 24,
            "marginBottom": 24,
          }
        }
      >
        Please check your e-mail's inbox for the password reset code and fill it in below.
      </Text>
      <TextInput
        autoCapitalize="characters"
        autoFocus={false}
        blurOnSubmit={false}
        clearButtonMode="always"
        keyboardType="default"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Reset password code, example: A6HC13"
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#e5e5e5",
            "borderRadius": 4,
            "borderWidth": 1,
            "fontSize": 15,
            "height": 50,
            "marginBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        testID="LoginResetPasswordForm-TextInput-reset-password-code"
        textContentType="username"
        value=""
      />
      <TextInput
        autoCapitalize="none"
        autoFocus={false}
        blurOnSubmit={false}
        clearButtonMode="always"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Your new password"
        returnKeyType="done"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#e5e5e5",
            "borderRadius": 4,
            "borderWidth": 1,
            "fontSize": 15,
            "height": 50,
            "marginBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        testID="LoginResetPasswordForm-TextInput-password"
        textContentType="password"
        value=""
      />
      <View>
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="LoginResetPasswordForm-Button-update-password"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Save my new password
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
